### DISCUSSION TITLE
Side download Fails

### DISCUSSION THREAD

**Discussion by gtmark1** [aclap-dev/video-downloadhelper#19] [Help and Q&A]
### Your question

Hiya Paul and thank you for your rely in Google Groups. This format is unfamiliar to me so I hope to do it correctly. As this is a "new discussion" on this format I will paste the original inquiry first.

<html><body>

"gtm...@gmail.com" <gtmark1@gmail.com>: Jan 30 09:26PM -0800                           I was just updated to VDH vrs. 8.2 and attempted two downloads from Youtube  and got "side download failed" on each. Until this update I have had months  of no failed downloads except for the occasional license reinstall.   Using a Lenovo Thinkpad  T60 I believe with a 12th Gen Intel(R) Core(TM)  i7-1270P and W11. This just happened tonight with the announce update.
--
"gtm...@gmail.com" <gtmark1@gmail.com>: Jan 30 09:43PM -0800                           addendum to my previous post-     SideDownload error: [webm @ 000001f767f33100] Only VP8 or VP9 or AV1 video  and Vorbis or Opus audio and WebVTT subtitles are supported for WebM.  [out#0/webm @ 000001f767c9eec0] Could not write header (incorrect codec  parameters ?): Invalid argument [aost#0:1/copy @ 000001f767f98c40] Error  initializing output stream:
Paul <paulrouget31415@gmail.com>: Jan 30 10:14PM -0800                           Does it work better if you use the MP4 variant? (if it did offer you that  option)   Do you mind posting more details  here: https://github.com/aclap-dev/video-downloadhelper/discussions   - link to youtube video - version of the CoApp

</body>
</html>"[gtm...@gmail.com](mailto:gtm...@gmail.com)" <[gtmark1@gmail.com](mailto:gtmark1@gmail.com)>: Jan 30 09:26PM -0800

I was just updated to VDH vrs. 8.2 and attempted two downloads from Youtube
and got "side download failed" on each. Until this update I have had months
of no failed downloads except for the occasional license reinstall.
 
Using a Lenovo Thinkpad T60 I believe with a 12th Gen Intel(R) Core(TM)
i7-1270P and W11. This just happened tonight with the announce update.
"[gtm...@gmail.com](mailto:gtm...@gmail.com)" <[gtmark1@gmail.com](mailto:gtmark1@gmail.com)>: Jan 30 09:43PM -0800

addendum to my previous post-
 
 
SideDownload error: [webm @ 000001f767f33100] Only VP8 or VP9 or AV1 video
and Vorbis or Opus audio and WebVTT subtitles are supported for WebM.
[out#0/webm @ 000001f767c9eec0] Could not write header (incorrect codec
parameters ?): Invalid argument [aost#0:1/copy @ 000001f767f98c40] Error
initializing output stream:
Paul <[paulrouget31415@gmail.com](mailto:paulrouget31415@gmail.com)>: Jan 30 10:14PM -0800

Does it work better if you use the MP4 variant? (if it did offer you that
option)
 
Do you mind posting more details
here: https://github.com/aclap-dev/video-downloadhelper/discussions
 
- link to youtube video
- version of the CoApp

Well, that was a strange copy and paste!  Anywho, I did notice that there was an update to the CoApp, although I thought I previously read it was only for MDP downloads but I went ahead and installed it anyway. And I tried a different video and got the same result for WEBM as posted above. I am only getting choices now for MP4s and WEBMs at various resolutions- no MKVs which is what I always chose in the past because I noticed they were bigger in MBs (thinking better quality).  The MP4 downloads are working where I did have issues sometimes in the past- MKVs were reliable.  And it does seem the MP4 downloads, of the 3 I have done, are fast.  

Just for kicks, here is the latest video that WEBM did not work but the MP4 did using the latest configuration.

https://www.youtube.com/watch?v=IggctNDices

Video DownloadHelper
Version 8.2.0.17a1 (beta)
Browser locale: en-US
Production build
Built on 2024-01-30
Build options: browser=firefox, linuxlic=false, noyt=false, target=mozilla

Platform Win x86-64
Browser Mozilla Firefox 122.0

Found companion app: VdhCoApp 2.0.14
Companion app binary: C:\Program Files\DownloadHelper CoApp\vdhcoapp.ex

I always need to learn from others so thanx for your replies.

And I hope your response makes it back to Google Groups as someone has the same problem-

        tonyb <[wesman3838@gmail.com](mailto:wesman3838@gmail.com)>: Jan 31 05:34PM -0800

         Same deal as above person. any fix yet?

I also hope that GitHub sends a notification of responses.
Thank you again.

**Comment by gtmark1** [comment]
And it would be nice if we can edit after the fact when using GitHub.  I wanted to add I hope I get notified of any response here.

**Reply by paulrouget** [reply]
You can edit it :)

**Comment by gtmark1** [comment]
Thanx . I did find the 3 dots at the top. I sent a long page :)

**Comment by paulrouget** [comment]
Thanks for posting here :)

- MKV are not offered anymore as their quality can't be better than the MP4.
- WEBM on youtube are broken, it's a known issue, and something I'm actively trying to fix.

For now, we recommend the MP4 download.

**Comment by gtmark1** [comment]
Ok, thanks for the reply.

**Comment by GrampaWildWilly** [comment]
It is my understanding that webm is a type of mkv, that mkv is a more general format & webm is a subset of mkv.  Also, I don't believe (somebody might correct me if I'm wrong) that there is no difference in quality between mp4 & mkv.  The file size differences are just due to differences in the format.  If you play an mp4 at a given resolution side by side with an mkv of the same resolution, both downloads of the same video from YouTube, do you actually perceive any difference in their quality?  I have looked at the Windows file properties of such side-by-side mp4/mkv pairs & even though the bit rates are different, the playback shows no quality differences.  The images are sharp (or not, depending on the content & how it was uploaded) & the audio sounds the same.  So I have always assumed that a bitrate for an mp4 & a bitrate for the eqivalent mkv can't be compared.  The formats are different & the bitrates are calculated differently.  The only reliable comparison is the one you can make with your eyeballs.  On balance, I'd get the smaller file at any given resolution, just to get it faster & to save a bit of space on your disk.  Besides, YouTube content is generally compressed, mercilessly compressed.  The bit rates are low because they want to save disk space, given that they host an enormous amount of content.  So you're not missing anything if you get the mp4 from now on instead of the mkv.

**Comment by GrampaWildWilly** [comment]
I just checked your carbide saw blades video.  None of the webm variants would download.  I thought only the highest resolution variants wouldn't download, that webm variants at 1920x1080 & below would still work.  I was wrong.  The 1920x1080 mp4 downloaded fine, with audio all the way to the end.  At one point I thought that the audio had cut off.  But I needed to turn the volume up a bit.  There's a section in the middle where there is no dialog, only some music.  I suppose music is preferable to the noise of a saw cutting wood.

**Reply by gtmark1** [reply]
Gramps- Paul (one of the devs) responded to me and said 3 things- MKV is no longer supported, WEBM has a known issue, and VDH is recommending MP4 downloads now- all about Youtube anyway. Thanx for following up and hope you enjoyed Emerald's video. She is good at what she does. And yes- the videos are simple and use some "filler".

**Reply by GrampaWildWilly** [reply]
At the risk of disappointing you, I skimmed the video.  I didn't pay close enough attention to realize her name is Emerald.  I'm sure she's quite a gem.

**Reply by gtmark1** [reply]
I am not disappointed. She is one daughter of 2 and a younger son helping their father run a family sawmill in Northern PA. You might find some "family" empathy by watching her videos. I just appreciate her enthusiasm in participating in her Dad's business.
